pkg/helmcli/adder.go

```diff
@@ -0,0 +1,134 @@
+package helmcli
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gojekfarm/albatross/pkg/helmcli/flags"
+	"github.com/gojekfarm/albatross/pkg/logger"
+
+	"github.com/gofrs/flock"
+	"gopkg.in/yaml.v2"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const timeout time.Duration = 30 * time.Second
+
+type adder struct {
+	flags.AddFlags
+	settings *cli.EnvSettings
+}
+
+func (o *adder) Add(ctx context.Context) error {
```

Comment:
Why are we not returning the added repo struct as part of func signature?

Reply:
it won't contain any additional information than the input payload supplied.
If we want to specify the list of charts available in the the added/updated repository. We would need to explicitly read the index file. 
Currently, writes everything to disk and does not create any object with additional metadata.